### PR TITLE
Fix 44 - Detect binary (CBOR) and skip rules engine processing in the 0MQ even…

### PR DESCRIPTION
**Pull request by jpwhitemn**
…t subscriber; ugly but works for now

Signed-off-by: Jim White <jpwhite_mn@yahoo.com>

### CODE REVIEW

File: src/main/java/org/edgexfoundry/messaging/ZeroMQEventSubscriber.java [hunk 1]
```diff
@@ -33,119 +35,126 @@
...
+		Event event;
+		logger.info("Watching for new exported Event messages...");
+		try {
+			while (!Thread.currentThread().isInterrupted()) {
+				if (exportClient) {
+					if (serializedJava) { // supporting legacy Java export distro that shipped Serialized eventString
+						exportString = subscriber.recvStr();
+						event = toEvent(exportString);
+					} else { // supporting new Go export distro that ships JSON
+						exportBytes = subscriber.recv();
```
Comment (tobiasmo1) [inline src/main/java/org/edgexfoundry/messaging/ZeroMQEventSubscriber.java]: Seems like a decent functional start.
For more deterministic handling, is it feasible to access/deserialize the [ContentType ](https://github.com/edgexfoundry/go-mod-messaging/blob/master/pkg/types/message_envelope.go#L38) property of MessageEnvelope?

**Comment by tobiasmo1** [comment]
ARM build failure reportedly caused by missing nss component.

`Caused by: java.io.FileNotFoundException: /usr/lib/libnss3.so at sun.security.pkcs11.Secmod.initialize(Secmod.java:193) at sun.security.pkcs11.SunPKCS11.<init>(SunPKCS11.java:218)`

- [Recent reports](https://github.com/docker-library/openjdk/issues/289) find that OpenJDK alpine base image is missing nss component.

- This [Alpine 3.7-3.9 bug](https://bugs.alpinelinux.org/issues/10126) was recently filed. Seems the 3.6 image is likely also exhibiting the error.

Workaround is to add nss separately (via **[Dockerfile.aarch64](https://github.com/edgexfoundry/support-rulesengine/blob/master/docker-files/Dockerfile.aarch64#L33)**):
`RUN apk add --no-cache nss`

File: src/main/java/org/edgexfoundry/messaging/ZeroMQEventSubscriber.java (hunk 1 above)
Comment (jpwhitemn) [reply src/main/java/org/edgexfoundry/messaging/ZeroMQEventSubscriber.java]: Toby, we would need a non-binary message envelop that would tip off Java or other system of what is contained in the envelop.  Right now, everything is encoded in binary - so we'd have to fail the JSON unmarshalling and then decode the binary just to get the type.  That's a lot of work for no gain at this point in time.  Going forward, I would really like to explore an envelop that itself is not encoded/binary.

**Comment by jpwhitemn** [comment]
Thanks for digging out the nss lib error Toby.  That fix worked.  Looking for approval to merge.

**Comment by tobiasmo1** [comment]
@tsconn23 I think you may be only reviewer with write access to this repo.

File: src/main/java/org/edgexfoundry/messaging/ZeroMQEventSubscriber.java (hunk 1 above)
Comment (tsconn23) [reply src/main/java/org/edgexfoundry/messaging/ZeroMQEventSubscriber.java]: Or possibly a message bus that supports the use of headers on messages to capture this kind of metadata.

**Pull request merged**